handler/core/src/main/java/io/knotx/fragments/handler/action/InMemoryCacheActionFactory.java

```diff
@@ -147,4 +200,31 @@ private SourceDefinitions buildSourceDefinitions(ClientRequest clientRequest) {
         .expireAfterWrite(ttl, TimeUnit.MILLISECONDS)
         .build();
   }
+
+  private static String getActionLogLevel(JsonObject config) {
+    return config.containsKey(LOG_LEVEL_KEY) ? config.getString(LOG_LEVEL_KEY)
+        : ActionLogLevel.ERROR.getLevel();
+  }
+
+  private static void logDoAction(ActionLogger actionLogger, long startTime,
+      FragmentResult fragmentResult) {
+    actionLogger.doActionLog(TimeCalculator.executionTime(startTime), fragmentResult.getNodeLog());
+  }
+
+  private static void logCacheLookup(ActionLogger actionLogger, String cacheKey) {
+    actionLogger.info(CACHE_LOOKUP, cacheKey);
+  }
+
+  private static void logCacheHit(ActionLogger actionLogger, Object cachedValue) {
+    actionLogger.info(CACHE_HIT, cachedValue);
+  }
+
+  private static void logCacheMiss(ActionLogger actionLogger, Object computedValue) {
+    actionLogger.info(CACHE_MISS, computedValue);
+  }
+
+  private static void logCachePass(ActionLogger actionLogger, FragmentResult failedResult) {
+    actionLogger.info(CACHE_PASS,
```

Comment:
Can we log also node log here? + unit test

Reply:
Node log will be provided by a call to `logDoAction` in `callDoActionAndCache`